src/cerebellum.py

```diff
@@ -0,0 +1,114 @@
+"""Pathfinding Module"""
+import pdb
+from queue import Queue
+from typing import List, Optional
+from src.snake import Snake
+from src.board import Board
+from src.coordinate import Coordinate
+
+
+class Cerebellum:
+    """Find paths to given coordinates."""
+    def __init__(self, brain, selected_algorithm: Optional[str]) -> None:
+        self.selected_algorithm: str = selected_algorithm if selected_algorithm is not None else 'breadth_first'
+        self.brain = brain
+        self.me = self.brain.me
+        self.board: Board = self.brain.board
+
+    def get_path(self, coordinate: Optional[Coordinate] = None, coordinates: Optional[List[Coordinate]] = None) -> List[Coordinate]:
+        """Find path to given coordinate/coordinates based on pathfinder's selected algorithm."""
+
+        if coordinates is None:
+            if coordinate is None:
+                return []
+            coordinates = [coordinate]
+
+        path: List[Coordinate] = []
+
+        if self.selected_algorithm == 'breadth_first':
+            path = self.__get_breadth_first_path(coordinates)
+
+#        elif self.selected_algorithm == 'djikstra\'s':
+#            path = self.__get_djikstras_path(coordinates)
+
+#        elif self.selected_algorithm == 'a*':
+#            path = self.__get_a_star_path(coordinates)
+
+        return path
+
+    def __get_breadth_first_path(self, coordinates: List[Coordinate]) -> List[Coordinate]:
+        """Find path using breadth-first algorithm."""
+
+        #pick location from grid (self.me.head)
+        #expand that location by looking at neighbours
+        #any unvisited neighbours are added to "unexplored" and "visited" set
+
+        for coord in coordinates:
+            print(coord)
+
+        start = self.me.head
+
+        unexplored: Queue = Queue()
+        unexplored.put(start)
+        came_from: dict = {}
+        came_from[str(start)] = None
+
+        collision_coordinates = [coordinate for snake in self.board.snakes for coordinate in snake.coordinates]
```

Comment:
I would create collision coordinates in `__init__` I think, and just reference them here. Or perhaps even better in their own function call that uses all the data passed in? We'll need it in a few places.

Reply:
I can see why this could be valuable, but there are (or at least were) places in brain where collision_coordinates is modified to contain more/less coordinates (tails, head neighbours, etc), so I figured since collision_coordinates isn't necessarily the same everywhere it'd be simplest just to create it separately?